internal/providers/pluginfw/resources/clusters/resource_library.go

```diff
@@ -0,0 +1,237 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/databricks/databricks-sdk-go"
+)
+
+const libraryDefaultInstallationTimeout = 15 * time.Minute
+
+var _ resource.ResourceWithConfigure = &LibraryResource{}
+
+func ResourceLibrary() resource.Resource {
+	return &LibraryResource{}
+}
+
+func parseId(id string) (string, string) {
+	split := strings.SplitN(id, "/", 2)
+	if len(split) != 2 {
+		return "unknown", "unknown"
+	}
+	return split[0], split[1]
+}
+
+func readLibrary(ctx context.Context, w *databricks.WorkspaceClient, waitParams compute.Wait, libraryRep string, libraryExtended *LibraryExtended) diag.Diagnostics {
+	res, err := libraries.WaitForLibrariesInstalledSdk(ctx, w, waitParams, libraryDefaultInstallationTimeout)
+	if err != nil {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("failed to wait for library installation", err.Error())}
+	}
+
+	for _, v := range res.LibraryStatuses {
+		thisRep := v.Library.String()
+		if thisRep == libraryRep {
+
+			diags := converters.GoSdkToTfSdkStruct(ctx, v.Library, libraryExtended)
+
+			if diags.HasError() {
+				return diags
+			}
+
+			libraryExtended.ClusterId = types.StringValue(waitParams.ClusterID)
+
+			return nil
+		}
+	}
+	return diag.Diagnostics{diag.NewErrorDiagnostic("failed to find the installed library", fmt.Sprintf("failed to find %s on %s", libraryRep, waitParams.ClusterID))}
+}
+
+type LibraryExtended struct {
+	compute_tf.Library
+	ClusterId types.String `tfsdk:"cluster_id" tf:""`
+}
+
+type LibraryResource struct {
+	Client *common.DatabricksClient
+}
+
+func (r *LibraryResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "databricks_library_pluginframework"
+}
+
+func (r *LibraryResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Library",
+		Attributes: tfschema.ResourceStructToSchemaMap(LibraryExtended{}, func(c tfschema.CustomizableSchema) tfschema.CustomizableSchema {
+			return c
+		}),
+	}
+}
+
+func (r *LibraryResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if r.Client == nil {
+		r.Client = pluginfwcommon.ConfigureResource(req, resp)
+	}
+}
+
+func (r *LibraryResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	clusterId, libraryRep := parseId(req.ID)
+	waitParams := compute.Wait{
+		ClusterID: clusterId,
+		IsRefresh: true,
+	}
+	lib := LibraryExtended{}
+	resp.Diagnostics.Append(readLibrary(ctx, w, waitParams, libraryRep, &lib)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, lib)...)
+
```

Comment:
Do we need to call the `readLibrary` method in `ImportState`? My understanding is that we only need to set whatever attributes the `Read()` method depends on. Basically, parse the ID, and set the clusterId and library state in the resource. Can you try this out?

See https://developer.hashicorp.com/terraform/plugin/framework/resources/import:

> Resources can implement the ImportState method, which must either specify enough Terraform state for the Read method to refresh [resource.Resource](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource#Resource) or return an error.

This simplifies this method's implementation, removes redundant API requests, and minimizes need for refactoring of your resource implementation (the read logic can live in the `Read()` method).

Reply:
Yes it does seem like this is suboptimal in terms of efficiency but I tried doing this and I don't think we can always reverse this serialisation.

For example for `pypi` and `mvn`, we cannot deterministically parse them since the sub fields are just blindly concatenated together without any separator.

Reply:
As discussed offline, we are not going to support import for now for migration purposes. Will considering modifying the .String() function before introducing the importer